### PR TITLE
[vdc] user creation logic

### PR DESCRIPTION
**Pull request by akotlar**
Contains user creation logic. Obvious improvements: update tests, add bit of logic to add resources to multiple namespaces (or define which resources should go in which namespace): right now gcp-sa-key needs to be in batch-pods, and (I think?) also default namespace

Has some duplicated database creation logic with auth

Please let me know what you'd like changed!

cc @cseed, @danking, @johnc1231

### CODE REVIEW

**Comment by akotlar** [comment]
Bump. What else would you all like to add to this PR for it to go through?

File: vdc/hail-ci-build.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+set -ex
+
+wd=$cwd
+cd scripts/users && ./hail-ci-build.sh && cd $wd
```
Comment (danking) [inline vdc/hail-ci-build.sh]: these types of files are no longer used

File: vdc/scripts/users/Makefile [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	./run-locally.sh $(file) create
+
+delete-user:
+	./run-locally.sh $(file) delete
+
+secret:
+	./make-secret.sh $(user) $(pass) $(db) $(host) $(instance)
+
+users.yml:
+	kubectl get secret users -o json | jq '.data["users.yml"]' -r | base64 -D > users.yml
```
Comment (danking) [inline vdc/scripts/users/Makefile]: I don't think this secret exists

File: vdc/scripts/users/modify_users.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    is_developer=${stringarray[2]}
+    is_sa=${stringarray[3]}
+    namespace=${stringarray[4]}
+    echo "email $email"
+    echo "auth0_id $auth0_id"
+    echo "is_developer $is_developer"
+    echo "is_sa $is_sa"
+    echo "namespace $namespace"
+    make user email="$email" user_id="$auth0_id" is_dev=$is_developer is_sa=$is_sa namespace="$namespace" op=$op
+done <users.txt
```
Comment (danking) [inline vdc/scripts/users/modify_users.sh]: where does this file come from?

File: vdc/scripts/users/Makefile (hunk 2 above)
Comment (akotlar) [reply vdc/scripts/users/Makefile]: It does, just deleted to update with gsarma and didn't add back. It's back.

File: vdc/scripts/users/modify_users.sh (hunk 3 above)
Comment (akotlar) [reply vdc/scripts/users/modify_users.sh]: From the users.yml line. The list of user configurations is present in the secrets "users". This contains the a base64-encoded YAML string, containing the full resource definition for each user.

**Comment by akotlar** [comment]
> A couple notes. I'm not sure what this should ultimately look like. I think I want `hailctl create user`, which probably creates a batch that does all the creation? That avoids setting up tunnels to the sql server, etc.

Sure, if that's the way we want this to work, I will modify to do that. Seems more elegant than having tunnels, which predates our current deployment solution.

**Comment by danking** [comment]
I think we should have a design discussion with Cotton before embarking on work as I'm sure he'll have input.

**Comment by tpoterba** [comment]
reopen when ready for review?

**Pull request closed without merging**